src/main/java/king/task/Deadline.java

```diff
@@ -0,0 +1,57 @@
+package king.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```

Comment:
The variable name "by" is too short and not sufficiently descriptive. It should be renamed to something more explanatory, such as "deadlineDateTime" or "dueDateTime".